sql/src/main/java/io/crate/executor/transport/TransportShardUpsertAction.java

```diff
@@ -459,31 +469,36 @@ private void shardIndexOperationOnReplica(ShardUpsertRequest request,
     void processGeneratedColumns(final DocTableInfo tableInfo,
                                  Map<String, Object> updatedColumns,
                                  Map<String, Object> updatedGeneratedColumns,
-                                 boolean validateExpressionValue) {
-        processGeneratedColumns(tableInfo, updatedColumns, updatedGeneratedColumns, validateExpressionValue, null);
+                                 boolean validateConstraints) {
+        processGeneratedColumns(tableInfo, updatedColumns, updatedGeneratedColumns, validateConstraints, null);
     }
 
     private void processGeneratedColumns(final DocTableInfo tableInfo,
                                          Map<String, Object> updatedColumns,
                                          Map<String, Object> updatedGeneratedColumns,
-                                         boolean validateExpressionValue,
+                                         boolean validateConstraints,
                                          @Nullable GetResult getResult) {
         SymbolToFieldExtractorContext ctx = new SymbolToFieldExtractorContext(functions, updatedColumns);
 
         for (GeneratedReferenceInfo referenceInfo : tableInfo.generatedColumns()) {
             // partitionedBy columns cannot be updated
             if (!tableInfo.partitionedByColumns().contains(referenceInfo)) {
                 Object givenValue = updatedGeneratedColumns.get(referenceInfo.ident().columnIdent().fqn());
-                if ((givenValue != null && validateExpressionValue)
+                if ((givenValue != null && validateConstraints)
                     ||
                     generatedExpressionEvaluationNeeded(referenceInfo.referencedReferenceInfos(), updatedColumns.keySet())) {
                     // at least one referenced column was updated, need to evaluate expression and update column
                     Function<GetResult, Object> extractor = SYMBOL_TO_FIELD_EXTRACTOR.convert(referenceInfo.generatedExpression(), ctx);
                     Object value = extractor.apply(getResult);
+
+                    if (validateConstraints) {
+                        ConstraintsValidator.validate(givenValue, referenceInfo);
```

Comment:
how does this work? It's in a ´if givenValue != null´ block but is validating that it shouldn't be null?